new: added DELETE api for entities

sadiqkhoja opened the pull request:
Closes #803 

#### What has been done to verify that this works as intended?

Integration tests

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

No

#### Does this change require updates to the API documentation? If so, please update docs/api.md as part of this PR.

We have a separate PR for that

#### Before submitting this PR, please make sure you have:

- [x] run `make test-full` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code from external sources are properly credited in comments or that everything is internally sourced

ktuite reviewed (commented):
Looks good but I think we should have some more tests

Please add:
- Use this deletion mechanism in test above called `should return metadata of the entities of the dataset - only deleted`
- Test that you can't access the deleted entity (get a 404) once its deleted, unless it's through the `entities?deleted=true` endpoint.

Maybe add:
- Test that it's not included in the CSV attachment (I'm not sure if that should go in this API test area or in the CSV export/attachment area. There are some other things that are not part of this PR I think we should also test in the CSV attachments, like that the updated values are getting passed through.) Made an issue https://github.com/getodk/central-backend/issues/842 to capture these thoughts.
- Eventually test that it IS included in the entity export? Or can be included? In the same way deleted submissions can be included?

sadiqkhoja commented:
Updated `should return metadata of the entities of the dataset - only deleted` and added more assertion for `should delete an Entity`

----

> Eventually test that it IS included in the entity export? Or can be included? In the same way deleted submissions can be included?

can be done as part of #801
